test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -30,6 +30,9 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/onsi/gomega"
+
```

Comment:
Ah, this is just gomega, but still. Use cmp instead.

Reply:
And you can use `k8s.io/apimachinery/pkg/util/wait`

Reply:
`util/wait` is awful for use in tests. All of the arguments from https://github.com/kubernetes/contributor-site/blob/bb1c87d320aacacc6ec5dd23d312fc59f93e3c6d/content/en/blog/2023/e2e-testing-best-practices.md#polling-and-timeouts also apply here.

I don't want to reimplement `gomega.Eventually`. If you insist, then I'll use `util/wait` and pretend that I don't know better.

Reply:
For E2E, I would agree, since we already use gomega intensively.

@aojea what is your take on gomega in integration tests vs the `wait` package?

> `util/wait` is awful for use in tests

Can we make it better?

Reply:
`util/wait` is used in production code. Adding things to it that are only needed for testing (like remembering past state to produce better error messages in case of a timeout) doesn't make sense. The API also would have to be changed, because a callback that just returns "bool, error" is not informative enough (why not ready?).

We would need two versions of it, and then we are back at "let's reimplement gomega.Eventually".

Reply:
> @aojea what is your take on gomega in integration tests vs the wait package?

Since last year I started to analyze this topic from a more objective angle and I realized that my bias to gomega was not to the Eventually method, was because of the BDD DSL patterns, that is always common on development that differentiate between unit test and integration/e2e test. Typically on companies that make the division between SW and QA engineers. I really don't like that division and I'm totally against it, a developer should cover the end to end, the feature and all the testing ( unit , integration and e2e test, ...), we also assume that in our KEPs

I really think we should avoid a BDD DSL pattern in kubernetes because of the reasons explained in the golang FAQ https://go.dev/doc/faq, it causes friction and I can't see any benefit , it is harder to debug also ...

> A related point is that testing frameworks tend to develop into mini-languages of their own, with conditionals and controls and printing mechanisms, but Go already has all those capabilities; why recreate them? We'd rather write tests in Go; it's one fewer language to learn and the approach keeps the tests straightforward and easy to understand.

So good so far, sorry for the long history, we come to a polling function for asynchronous assertion, that is `gomega.Eventually` vs `wait.Poll`.
I personally never find any issue using `wait.PollImmediate` but I can not generalize my opinion, Patrick highlighted some of the benefits of using `gomega.Eventually`in https://github.com/kubernetes/contributor-site/blob/bb1c87d320aacacc6ec5dd23d312fc59f93e3c6d/content/en/blog/2023/e2e-testing-best-practices.md#polling-and-timeouts , so my conclusion is that is a code maintainers choice to decide which function to use for an asynchronous checks and live with the pros and cons of their decision, at the end, once the code is merged, the persons responsible to maintain the code have to live with that decision.